File: hw/nase/u3/hw21AnimalFarm/JavaScript/model/interfaces.js
class IMeatAnimal {
    cut() { throw new Error("The 'cut()' method must be implemented.."); }
    sendToSlaughterHouse(slaughterHouse) { throw new Error("Método 'sendToSlaughterHouse()' must be implemented."); }
}

class IProduceAnimal {
    procedure() { throw new Error("Método 'procedure()' must be implemented."); }
    measureQuantity(unit, quantity) { throw new Error("Método 'measureQuantity()' must be implemented."); }
}

module.exports = { IMeatAnimal, IProduceAnimal };

File: hw/nase/u3/hw21AnimalFarm/Python/model/interfaces.py
from abc import ABC, abstractmethod
from typing import List

class IMeatAnimal(ABC):
    @abstractmethod
    def cut(self) -> List['Cut']:
        pass

    @abstractmethod
    def send_to_slaughter_house(self, slaughter_house: 'SlaughterHouse') -> None:
        pass

class IProduceAnimal(ABC):
    @abstractmethod
    def procedure(self) -> str:
        pass

    @abstractmethod
    def measure_quantity(self, unit: str, quantity: float) -> None:
        pass